dual_net.py

```diff
@@ -162,21 +172,30 @@ def __call__(self, inp):
         Gets a feed-forward prediction for a batch of input boards of shape set
         during initialization.
         """
+        move_legality_mask = np.zeros(shape=(inp.shape[0], self.action_size))
+        for i in range(inp.shape[0]):
+            move_legality_mask[i] = self.env.get_legality_mask(inp[i])
         policy, value = self.sess.run([self.policy_predict, self.value_predict],
-                                      feed_dict={self.board_placeholder: inp})
+                                      feed_dict={self.board_placeholder: inp,
+                                                 self.move_legality_mask: move_legality_mask})
         return policy, value
 
-    def train(self, boards, pi, z):
+    def train(self, states, pi, z, legality_mask):
```

Comment:
If we have the environment as a field of the net then we don't need this legality_mask I think